server/neptune/workflows/internal/activities/execute.go

```diff
@@ -1,12 +1,99 @@
 package activities
 
-import "context"
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
 
-type executeCommandActivities struct{}
+	"github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/runtime/cache"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/steps"
+)
+
+type executeCommandActivities struct {
+	// Version Cache ensures the terraform version exists
+	VersionCache cache.ExecutionVersionCache
+
+	// DefaultTFVersion is the default configured version for atlantis is the tfVersion is not supplied
+	DefaultTFVersion *version.Version
+
+	// TerraformBinDir is the directory where Atlantis downloads Terraform binaries.
+	TerraformBinDir string
+}
+
+type ExecuteCommandResponse struct {
+	Output string
+	Error  error
+}
 
 type ExecuteCommandRequest struct {
+	Step           steps.Step
+	Path           string
+	CustomEnvVars  map[string]string
+	DefaultEnvVars map[string]string
+
+	// TerraformVersion is the version of terraform we should use when executing
+	// commands for this project. This can be set to empty in which case we will
+	// use the default Atlantis terraform version.
+	TerraformVersion string
 }
 
-func (t *terraformActivities) ExecuteCommand(ctx context.Context, request ExecuteCommandRequest) error {
-	return nil
+func (t *executeCommandActivities) ExecuteCommand(ctx context.Context, request ExecuteCommandRequest) ExecuteCommandResponse {
+
+	var err error
+	terraformVersion := t.DefaultTFVersion
+	if request.TerraformVersion != "" {
+		terraformVersion, err = version.NewVersion(request.TerraformVersion)
+		if err != nil {
+			return ExecuteCommandResponse{
+				Output: "",
+				Error:  errors.Wrapf(err, "parsing version %s", request.TerraformVersion),
+			}
+		}
+	}
+
+	// Ensure version exists
+	_, err = t.VersionCache.Get(terraformVersion)
+	if err != nil {
+		return ExecuteCommandResponse{
+			Output: "",
+			Error:  errors.Wrapf(err, "getting version %s", terraformVersion),
+		}
+	}
+
+	// defaultTfVersion and the TerraformBinDir is only configured in the worker
+	terraformEnvVars := map[string]string{
+		"ATLANTIS_TERRAFORM_VERSION": terraformVersion.String(),
```

Comment:
Does our usecase rely on `ATLANTIS_TERRAFORM_VERSION`? If not, can we save downloading for TF activities?

Reply:
Yeah I originally had it to support the terraform env vars but since we don't need it, I have removed it.